src/plugins/dashboard/public/dashboard_app/dashboard_app.test.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { render, waitFor } from '@testing-library/react';
+import { MemoryHistory, createMemoryHistory } from 'history';
+import React, { useEffect } from 'react';
+import type { DashboardRendererProps } from '../dashboard_container/external_api/dashboard_renderer';
+import { buildMockDashboard } from '../mocks';
+import { DashboardApp } from './dashboard_app';
+
+import * as dashboardRendererStuff from '../dashboard_container/external_api/lazy_dashboard_renderer';
+import { DashboardApi } from '..';
+
+/* These tests circumvent the need to test the router and legacy code
+/* the dashboard app will be passed the expanded panel id from the DashboardRouter through mountApp()
+/* @link https://github.com/elastic/kibana/pull/190086/
+*/
+
+describe('Dashboard App', () => {
+  const mockDashboard = buildMockDashboard();
+  let mockHistory: MemoryHistory;
+  let historySpy: jest.SpyInstance;
+  let mockDashboardApiExpandedPanel: string | undefined;
+  const expandPanelSpyInRouter = jest.spyOn(mockDashboard, 'expandPanel');
```

Comment:
This isn't in the router :) I would just call this `expandPanelSpy`

```suggestion
  const expandPanelSpy = jest.spyOn(mockDashboard, 'expandPanel');
```

Reply:
I guess my point in the naming is that it's mimicking the DashboardRouter pulling the params and then passing it to the dashboard app and the dashboard renderer. I'd like to keep the naming to differentiate it from the history spy that is within the expandedPanelId subscription if it's ok with you

Reply:
> I guess my point in the naming is that it's mimicking the DashboardRouter pulling the params 

Is it? 🤔 Your spying on the **dashboard container** `expandPanel` method, and the container doesn't exist in the router. So I'm not sure what this has to do with the router in this case? I'm fine keeping it if I'm misunderstanding, but I'm still a bit confused what you mean :)

Reply:
[6eed21c](https://github.com/elastic/kibana/pull/190086/commits/6eed21c1eb2402ceb3972b67eb94e6eb44eb5996) thanks for pairing!